scc/node/node.go

```diff
@@ -0,0 +1,44 @@
+package node
+
+import (
+	"fmt"
+
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+)
+
+// Node is a node in the Sonic Certification Chain. It is responsible for
+// handling the progression of the chain by responding to new block statements
+// and creating new certificates.
+type Node struct {
+	store Store
+	state State
+}
+
+// NewNode creates a new node with the given store and committee. The provided
+// committee is expected to be the active committee of the current block.
```

Comment:
expected ? There is no way of quickly verifying this, right ? just brainstorming here, but since this method is not something that would be called many times, it could perform a heavy task such as verifying the with a couple of RPC calls that the committee is the active one, no ?

Reply:
The source for this committee should be the local node's DB, which is -- from the perspective of the local node -- a trusted source, filled through one of many possible synchronization procedures:
- a full-sync from block 0
- a G-file based sync from a given block height
- a light-client protocol based sync

Any of these procedures should be carried out to fill the content of the local DB, before this new `Node` is created.

Also, this is not the final version of this function, as the factory parameters will evolve over the course of implementing this feature.

Reply:
aahh, I thought this node would be the one holding the state for the client to sync in.